File: simple/src/main/java/com/tuuzed/adapter/simple/MainActivity.java
package com.tuuzed.adapter.simple;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.TextView;

import com.tuuzed.adapter.RecyclerViewAdapter;
import com.tuuzed.adapter.SimpleItemProvider;
import com.tuuzed.adapter.ViewHolder;

public class MainActivity extends AppCompatActivity {

    private RecyclerViewAdapter mAdapter;
    private RecyclerView mRecyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRecyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        mAdapter = new RecyclerViewAdapter(mRecyclerView);
        mAdapter.register(String.class, new SimpleItemProvider<String>(R.layout.item_string) {
            @Override
            public void onBindViewHolder(@NonNull ViewHolder holder, @NonNull String item, int position) {
                holder.$(R.id.text, TextView.class).setText(item + ": " + "STRING");
            }
        });
        mAdapter.register(Integer.class, new SimpleItemProvider<Integer>(R.layout.item_integer) {
            @Override
            public void onBindViewHolder(@NonNull ViewHolder holder, @NonNull Integer item, int position) {
                holder.$(R.id.text, TextView.class).setText(item + ": " + "INTEGER");
            }
        });
        mAdapter.register(Long.class, new SimpleItemProvider<Long>(R.layout.item_long) {
            @Override
            public void onBindViewHolder(@NonNull ViewHolder holder, @NonNull Long item, int position) {
                holder.$(R.id.text, TextView.class).setText(item + ": " + "LONG");
            }
        });
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        for (int i = 0; i < 50; i++) {
            mAdapter.items().add((long) i);
            mAdapter.items().add(String.valueOf(i));
            mAdapter.items().add(i);
        }
        mAdapter.notifyDataSetChanged();
    }
}

File: sample/src/main/java/com/tuuzed/recyclerview/adapter/sample/MainActivity.kt
package com.tuuzed.recyclerview.adapter.sample

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.text.InputType
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.tuuzed.recyclerview.adapter.RecyclerViewAdapter
import com.tuuzed.recyclerview.adapter.prefs.dsl.*
import com.tuuzed.recyclerview.adapter.prefs.withPrefs
import kotlinx.android.synthetic.main.main_act.*

class MainActivity : AppCompatActivity() {

    private lateinit var cxt: Context
    private lateinit var listAdapter: RecyclerViewAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        cxt = this
        setContentView(R.layout.main_act)
        recyclerView.layoutManager = LinearLayoutManager(this)
        listAdapter = RecyclerViewAdapter.with(recyclerView).withPrefs().dslPref {
            clickable {
                title = "Feed List"
                summary = "Feed List"
                click = { _, _ ->
                    startActivity(Intent(cxt, FeedListActivity::class.java))
                    true
                }
            }
            category {
                title = "Prefs"
            }
            general {
                title = "General"
                summary = "General#summary"
            }
            editText {
                title = "EditText"
                summary = "EditText#summary"
                inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_FLAG_MULTI_LINE
            }
            editText {
                title = "EditText"
                summary = "EditText#summary"
                allowEmpty = true
                inputType = InputType.TYPE_CLASS_TEXT or InputType.TYPE_TEXT_FLAG_MULTI_LINE
            }
            divider {}
            checkBox {
                title = "CheckBox"
                summary = "CheckBox#summary"
            }
            switch {
                title = "Switch"
                summary = "Switch#summary"
            }
            radio {
                title = "Radio"
                summary = "Radio#summary"
            }
            divider {}
            singleList {
                title = "SingleList"
                summary = "SingleList#summary"
                itemsLoader = {
                    it(arrayListOf(
                            "Single Item 1",
                            "Single Item 2",
                            "Single Item 3",
                            "Single Item 4"
                    ))
                }
            }
            multiList {
                title = "MultiList"
                summary = "MultiList#summary"
                itemsLoader = {
                    it(arrayListOf(
                            "Multi Item 1",
                            "Multi Item 2",
                            "Multi Item 3",
                            "Multi Item 4"
                    ))
                }
            }
        }
        listAdapter.notifyDataSetChanged()
    }
}
